x-pack/legacy/plugins/snapshot_restore/public/app/components/policy_form/steps/step_settings.tsx

```diff
@@ -0,0 +1,450 @@
+                  selectIndicesMode === 'list' ? (
+                    <EuiFlexGroup justifyContent="spaceBetween">
+                      <EuiFlexItem grow={false}>
+                        <FormattedMessage
+                          id="xpack.snapshotRestore.policyForm.stepSettings.selectIndicesLabel"
+                          defaultMessage="Select indices"
+                        />
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        <EuiLink
+                          onClick={() => {
+                            setSelectIndicesMode('custom');
+                            updatePolicyConfig({ indices: indexPatterns.join(',') });
+                          }}
+                        >
+                          <FormattedMessage
+                            id="xpack.snapshotRestore.policyForm.stepSettings.indicesToggleCustomLink"
+                            defaultMessage="Use index patterns"
+                          />
+                        </EuiLink>
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  ) : (
+                    <EuiFlexGroup justifyContent="spaceBetween">
+                      <EuiFlexItem grow={false}>
+                        <FormattedMessage
+                          id="xpack.snapshotRestore.policyForm.stepSettings.indicesPatternLabel"
+                          defaultMessage="Index patterns"
+                        />
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        <EuiLink
+                          onClick={() => {
+                            setSelectIndicesMode('list');
+                            updatePolicyConfig({ indices: cachedIndices });
+                          }}
+                        >
+                          <FormattedMessage
+                            id="xpack.snapshotRestore.policyForm.stepSettings.indicesToggleListLink"
+                            defaultMessage="Select indices"
+                          />
+                        </EuiLink>
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  )
+                }
+                helpText={
+                  selectIndicesMode === 'list' ? (
+                    <FormattedMessage
+                      id="xpack.snapshotRestore.policyForm.stepSettings.selectIndicesHelpText"
+                      defaultMessage="{count} {count, plural, one {index} other {indices}} will be backed up. {selectOrDeselectAllLink}"
+                      values={{
+                        count: config.indices && config.indices.length,
+                        selectOrDeselectAllLink:
+                          config.indices && config.indices.length > 0 ? (
+                            <EuiLink
+                              onClick={() => {
+                                indicesOptions.forEach((option: Option) => {
```

Comment:
This seems to work because the component is re-rendered, but I think it would be better to update the state with `setIndicesOptions()`

Reply:
Cool. Can you add a comment pointing to it so if we come back later we can check if it has been fixed?